### PR TITLE
Update validator for notifications

### PR DESCRIPTION
**Pull request by drhenner**
The tests I added failed.  Changing the validator fixed the issues.

### CODE REVIEW

File: spec/models/notification_spec.rb [hunk 1]
```diff
@@ -25,6 +25,30 @@
...
 
+    context 'skus and users can be associated to a notification' do
+      let(:duplicate_id) { 1000000009 }
+      let!(:sku)  { FactoryGirl.create(:sku, id: duplicate_id) }
+      let!(:user) { FactoryGirl.create(:user, id: duplicate_id) }
+
+      it 'should allow different models to be associated' do
+        sku_notification = create(:sku_notification, email: 'qwerty21@gmail.com', notifiable: sku)
+        expect(sku_notification.valid?).to be true
+        notification = build(:user_notification, email: 'qwerty21@gmail.com', notifiable: user)
```
Comment (Jellyfishboy) [inline spec/models/notification_spec.rb]: @drhenner  should really be using let method to create records to ensure the DatabaseCleaner wipes the db between specs.

File: spec/models/notification_spec.rb [hunk 2]
```diff
@@ -25,6 +25,30 @@
...
-    it { expect(create(:sku_notification)).to validate_uniqueness_of(:email).scoped_to(:notifiable_id).with_message('notification has already been created.') }
+    it { expect(create(:sku_notification)).to validate_uniqueness_of(:email).scoped_to([:notifiable_id, :notifiable_type, :sent_at]).with_message('notification has already been created.') }
 
+    context 'skus and users can be associated to a notification' do
+      let(:duplicate_id) { 1000000009 }
+      let!(:sku)  { FactoryGirl.create(:sku, id: duplicate_id) }
+      let!(:user) { FactoryGirl.create(:user, id: duplicate_id) }
+
+      it 'should allow different models to be associated' do
+        sku_notification = create(:sku_notification, email: 'qwerty21@gmail.com', notifiable: sku)
```
Comment (Jellyfishboy) [inline spec/models/notification_spec.rb]: @drhenner please use let helper method.

File: spec/models/notification_spec.rb [hunk 3]
```diff
@@ -25,6 +25,30 @@
...
+        expect(notification.save).to be true
+      end
+    end
+
+    context 'skus and users can be associated to a notification' do
+      let(:user) { FactoryGirl.create(:user) }
+
+      it 'should allow a second association if the first was sent years ago' do
+        expect(create(:user_notification, email: 'qwerty22@gmail.com', notifiable: user, sent_at: 2.years.ago, created_at: 3.years.ago).valid?).to be true
+        notification = build(:user_notification, email: 'qwerty22@gmail.com', notifiable: user)
```
Comment (Jellyfishboy) [inline spec/models/notification_spec.rb]: @drhenner please use let helper method

**Comment by drhenner** [comment]
@Jellyfishboy I agree with using the `let` syntax being cleaner but I have never heard of:

        should really be using let method to create records to ensure the DatabaseCleaner wipes the db between specs. 

Where can I find docs about DatabaseCleaner having issues without the let syntax?

**Comment by drhenner** [comment]
Khun phuut phassa Thai Mai?  Phom Yui ti Krung Thep Nung pii.  But that was ten years ago.

**Comment by Jellyfishboy** [comment]
Yes I can speak Thai, lived here for 2 years haha.

ใช่พูดภาษาไทยได้ อยู่บ้านประเทศไทยสองปี

File: app/models/notification.rb [hunk 4]
```diff
@@ -22,6 +22,9 @@ class Notification < ActiveRecord::Base
 
   belongs_to :notifiable, polymorphic: true
 
-  validates :email,         presence: { message: 'is required' }, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i }, uniqueness: { scope: [:notifiable_id, :notifiable_type, :sent_at], message: 'notification has already been created.' }
+  validates :email,         presence:   { message: 'is required' },
+                            format:     { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i },
+                            uniqueness: { scope: [:notifiable_id, :notifiable_type, :sent_at],
+                                          message: 'notification has already been created.' }
```
Comment (Jellyfishboy) [inline app/models/notification.rb]: @drhenner Thanks, love the formatting 👍

**Comment by Jellyfishboy** [comment]
> Where can I find docs about DatabaseCleaner having issues without the let syntax?

I shall dig out my source article, around here somewhere. I original had the issue before, hence the custom DatabaseCleaner config in `spec/support`.

**Pull request merged**